packages/block-library/src/block/test/edit.native.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * External dependencies
+ */
+import {
+	getEditorHtml,
+	initializeEditor,
+	fireEvent,
+	waitFor,
+	within,
+} from 'test/helpers';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import fetchRequest from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { registerCoreBlocks } from '../..';
+
+const getMockedReusableBlock = ( id ) => ( {
+	content: {
+		raw: `
+    <!-- wp:heading -->
+    <h2>First Reusable block</h2>
+    <!-- /wp:heading -->
+    
+    <!-- wp:paragraph -->
+    <p><strong>Bold</strong> <em>Italic</em> <s>Striked</s> Superscript<sup>(1)</sup> Subscript<sub>(2)</sub> <a href="http://www.wordpress.org" target="_blank" rel="noreferrer noopener">Link</a></p>
+    <!-- /wp:paragraph -->
+    
+    !-- wp:heading {"level":4} -->
+    <h4>List</h4>
+    <!-- /wp:heading -->
+    
+    <!-- wp:list -->
+    <ul><li>First Item</li><li>Second Item</li><li>Third Item</li></ul>
+    <!-- /wp:list -->
+`,
+	},
+	id,
+	title: { raw: `Reusable block - ${ id }` },
+	type: 'wp_block',
+} );
+
+beforeAll( () => {
+	// Register all core blocks
+	registerCoreBlocks();
+} );
+
+afterAll( () => {
+	// Clean up registered blocks
+	getBlockTypes().forEach( ( block ) => {
+		unregisterBlockType( block.name );
+	} );
+} );
+
+describe( 'Reusable block', () => {
+	it( 'renders warning when the block does not exist', async () => {
+		// We have to use different ids because entities are cached in memory.
+		const id = 1;
+		const initialHtml = `<!-- wp:block {"ref":${ id }} /-->`;
+		const endpoint = `/wp/v2/blocks/${ id }`;
+
+		const { getByA11yLabel } = await initializeEditor( {
+			initialHtml,
+		} );
+
+		const reusableBlock = await waitFor( () =>
+			getByA11yLabel( /Reusable block Block\. Row 1/ )
+		);
+
+		// Wait for the block to be fetched.
+		const reusableBlockRequest = await waitFor( () =>
+			fetchRequest.mock.calls.find( ( call ) =>
+				call[ 0 ].path.startsWith( endpoint )
+			)
+		);
```

Comment:
Do we need use `waitFor` or could this be simplified to the following? What asynchronous work is occurring here? 
```suggestion
		const reusableBlockRequest = fetchRequest.mock.calls.find( ( call ) =>
			call[ 0 ].path.startsWith( endpoint )
		);
```

Reply:
The request is asynchronous because the `api-fetch` mock returns a Promise so technically speaking we should wait, however, as this part is right after the `waitFor` of the reusable block, by the time is executed it will have the request.

Btw, I based this part on [this example](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor).

Reply:
I confirm that it works without `waitFor` so I'll remove it, thanks for the suggestion 🙇 !

Reply:
> The request is asynchronous because the `api-fetch` mock returns a Promise so technically speaking we should wait, however, as this part is right after the `waitFor` of the reusable block, by the time is executed it will have the request.
> 
> Btw, I based this part on [this example](https://testing-library.com/docs/dom-testing-library/api-async/#waitfor).

Ah, OK. 

> I confirm that it works without `waitFor` so I'll remove it, thanks for the suggestion 🙇 !

That is what I thought might be the case, because we aren't actually doing anything with the return value here. We are merely assigning the value to a `const`. The value could be a pending or resolved `Promise`, but its status doesn't really matter for this specific line. There is no need to await the resolution on this line specifically. 

Currently, all we [assert is that the value exists](https://github.com/WordPress/gutenberg/blob/962f87f9e29f69b553c098f429d5f56605630e69/packages/block-library/src/block/test/edit.native.js#L88), which will likely be a pending or unresolved `Promise`. 

As I talk through this again, my only concern from removing `await` here would be that it might introduce `Promise` resolutions and callbacks occurring _after_ the test completes, which might output a warning. I suppose as long as that does not occur, removing `await` should be fine.

Reply:
>Currently, all we assert is that the value exists, which will likely be a pending or unresolved Promise.
Not sure, I think the Promise will get resolved by the Reusable block so most likely it won't be left pending or unresolved. Well, it could happen if when we check if the mock was called, it wasn't,  but in this case the test will fail 😄 .

>As I talk through this again, my only concern from removing await here would be that it might introduce Promise resolutions and callbacks occurring after the test completes, which might output a warning. I suppose as long as that does not occur, removing await should be fine.

I did a quick check and I confirm that it doesn't so look safe to apply your suggestion 🎊 .

Reply:
I applied this suggestion in this [commit](https://github.com/WordPress/gutenberg/pull/32038/commits/bbc4994e00f066de3852fa22abf3deb8c99d1a40).